tests/utils/httpd.py

```diff
@@ -30,20 +31,32 @@ def end_headers(self):
 
 
 class StaticFileServer:
-    __server_lock = threading.Lock()
-
     def __init__(self, handler="etag"):
-        handler_class = ETagHandler if handler == "etag" else ContentMD5Handler
 
-        self.__server_lock.acquire()
-        self.httpd = HTTPServer(("localhost", 8000), handler_class)
+        handler_class = ETagHandler if handler == "etag" else ContentMD5Handler
+        self.bind_port_by_any_means(handler_class)
+
+    def bind_port_by_any_means(self, handler_class):
```

Comment:
is it used in tests?

Reply:
Yes, problem with static server, is that even that we close and shutdown socket, then server, the port is still binded for some unspecified amount of time. We can solve it two ways:
- each time we use server draw different port
- try to bind it few times 
Me and @efiop  decided to go with the second approach. 
Do you think I should change that?

Reply:
I'm wonder if there are libraries that help you stub/mock HTTP API, some thing similar to https://github.com/bblimke/webmock . It a good practice to mock/stub HTTP.

Reply:
Good point. Do you think we should solve it in current tak, or make a new one?